ls1intum/Artemis
# `Grading`: Fix course statistics to show correct numbers in tooltips

### Checklist
#### General

#### Client

### Motivation and Context
Closes https://github.com/ls1intum/Artemis/issues/6323

### Description
- When hiding a score line from the "Performance in Exercises" chart, the data is now removed instead of set to 0
- This way, the tooltip does not show incorrect numbers anymore
- The data was also modified to not be off by 1 anymore (was needed before to not show the tooltip for "hidden" values)

### Steps for Testing

Prerequisites:
- A course in which the user participated in exercises

1. Log in to Artemis
2. Navigate to the Course
3. Navigate to Statistics
4. Scroll to the "Performance in Exercises" Chart
5. Hover over the chart (NOT directly over a data point)
6. You should see all three scores for an exercise
7. Toggle off a score in the legend
8. Hover over the chart again
9. You now should only see the two scores that are still toggled on

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage





#### Client

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [exercise-scores-chart.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5390/latest/artifact/shared/Coverage-Report-Client-Tests/app/overview/visualizations/exercise-scores-chart/exercise-scores-chart.component.ts.html) | 95% | ✅ ❌ |

### Screenshots
Before the Bugfix:
![image](https://user-images.githubusercontent.com/123463270/223002668-5eea368e-ca01-4ed9-9b3e-87c2bb8f0edd.png)

After the Bugfix:

![image](https://github.com/ls1intum/Artemis/assets/118574504/74621297-dc34-4c8f-ad08-05404be9dbaf)
![image](https://github.com/ls1intum/Artemis/assets/118574504/0d03c4c8-52cf-47ae-b01d-9c8ed06dcf1d)
